### ISSUE TITLE
No `tui` option in version 0.1.19.3 on aarch64

### ISSUE THREAD

**Issue by tobsz** [haskell/ghcup-hs#843]
As the title says, there is no `ghcup tui` possible with ghcup's current version 0.1.19.3 on aarch64.
To double check, I just downloaded the binaries of 0.1.19.2 and 0.1.19.3 from `https://downloads.haskell.org/~ghcup/` and ran `ghcup tui` with both. 0.1.19.2 worked as expected/before, 0.1.19.3 did not.

**Debug Info**:
```
Architecture: aarch64
Platform: Linux UnknownLinux (it's ArchLinuxArm)
Version: 0.1.19.3
```

**Output of `ghcup tui` (0.1.19.3)**:
```
Invalid argument `tui'

Did you mean this?
    run

Usage: ghcup [(-v|--verbose) | --no-verbose] [(-c|--cache) | --no-cache] 
             [--metadata-caching SEC] [--metadata-fetching-mode <Strict|Lax>] 
             [-p|--platform PLATFORM] [--verify | (-n|--no-verify)] 
             [(-o|--offline) | --no-offline] [--gpg <strict|lax|none>] COMMAND

Available options:
  -v,--verbose             Enable verbosity (default: disabled)
  -c,--cache               Cache downloads in ~/.ghcup/cache (default: disabled)
  --metadata-caching SEC   How long the yaml metadata caching interval is (in
                           seconds), 0 to disable
  --metadata-fetching-mode <Strict|Lax>
                           Whether to fail on metadata download failure (Strict)
                           or fall back to cached version (Lax (default))
  -p,--platform PLATFORM   Override for platform (triple matching ghc tarball
                           names), e.g. x86_64-fedora27-linux
  -n,--no-verify           Disable tarball checksum verification (default:
                           enabled)
  --keep <always|errors|never>
                           Keep build directories? (default: errors)
  --downloader <curl|wget> Downloader to use (default: curl)
  -o,--offline             Don't do any network calls, trying cached assets and
                           failing if missing.
  --gpg <strict|lax|none>  GPG verification (default: none)
  -h,--help                Show this help text
  --version                Show version
  --numeric-version        Show the numeric version (for use in scripts)

Main commands:
  install                  Install or update GHC/cabal/HLS/stack
  test                     Run tests for a tool (if any) [EXPERIMENTAL!]
  set                      Set currently active GHC/cabal version
  unset                    Unset currently active GHC/cabal version
  rm                       Remove a GHC/cabal/HLS/stack version
  list                     Show available GHCs and other tools
  upgrade                  Upgrade ghcup
  compile                  Compile a tool from source
  whereis                  Find a tools location
  prefetch                 Prefetch assets
  gc                       Garbage collection
  run                      Run a command with the given tool in PATH

Other commands:
  debug-info               Show debug info
  tool-requirements        Show the requirements for ghc/cabal
  changelog                Find/show changelog
  config                   Show or set config

Nuclear Commands:
  nuke                     Completely remove ghcup from your system
```

**Comment by hasufell** [comment]
Found the bug. We use a too old cabal on arm64, which doesn't understand os/architecture conditionals in `cabal.project.release`: 

https://github.com/haskell/ghcup-hs/blob/1f220cd4883f24848b4a4bc10171494718894e55/.github/workflows/release.yaml#L84C13-L84C13

https://github.com/haskell/ghcup-hs/blob/1f220cd4883f24848b4a4bc10171494718894e55/cabal.project.release#L9

**Comment by hasufell** [comment]
This will require a new release, I guess.

**Comment by hasufell** [comment]
This is fixed in 0.1.19.4, which has just been released.
